lib/Agrammon/DB/Dataset.pm6

```diff
@@ -329,21 +329,21 @@ class Agrammon::DB::Dataset does Agrammon::DB {
                                                            AND data_instance IS NULL
                 RETURNING data_comment
             SQL
-
-            return if $ret.rows;
+            return $ret.rows if $ret.rows;
 
             $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $variable);
                 INSERT INTO data_new (data_dataset, data_var, data_comment)
                      VALUES          (dataset_name2id($2,$3), $4, $1)
                 RETURNING data_comment
                 SQL
+            return $ret.rows;
 
             # couldn't save comment
             die X::Agrammon::DB::Dataset::StoreInputCommentFailed.new(:$comment, :$variable) unless $ret.rows;
```

Comment:
This `die` is unreachable because it's after a `return`.

Reply:
Added the `return` at the wrong place, glad you caught it. Fixed.